ui/component/commentsReplies/view.jsx

```diff
@@ -1,127 +1,135 @@
   numDirectReplies: number, // Total replies for parentId as reported by 'comment[replies]'. Includes blocked items.
   isFetchingByParentId: { [string]: boolean },
-  onShowMore?: () => void,
   hasMore: boolean,
   supportDisabled: boolean,
+  doResolveUris: (Array<string>) => void,
+  onShowMore?: () => void,
 };
 
 function CommentsReplies(props: Props) {
   const {
     uri,
     parentId,
     fetchedReplies,
+    resolvedReplies,
     claimIsMine,
     myChannels,
     linkedCommentId,
-    commentingEnabled,
+    userCanComment,
     threadDepth,
     numDirectReplies,
     isFetchingByParentId,
-    onShowMore,
     hasMore,
     supportDisabled,
+    doResolveUris,
+    onShowMore,
   } = props;
 
   const [isExpanded, setExpanded] = React.useState(true);
+  const [commentsToDisplay, setCommentsToDisplay] = React.useState(fetchedReplies);
+  const isResolvingReplies = fetchedReplies && resolvedReplies.length !== fetchedReplies.length;
+  const alreadyResolved = !isResolvingReplies && resolvedReplies.length !== 0;
+  const canDisplayComments = commentsToDisplay && commentsToDisplay.length === fetchedReplies.length;
 
-  function showMore() {
-    if (onShowMore) {
-      onShowMore();
-    }
-  }
+  // Batch resolve comment channel urls
+  React.useEffect(() => {
+    if (!fetchedReplies || alreadyResolved) return;
 
-  // todo: implement comment_list --mine in SDK so redux can grab with selectCommentIsMine
-  function isMyComment(channelId: string) {
-    if (myChannels != null && channelId != null) {
-      for (let i = 0; i < myChannels.length; i++) {
-        if (myChannels[i].claim_id === channelId) {
-          return true;
-        }
-      }
-    }
-    return false;
-  }
+    const urisToResolve = [];
+    fetchedReplies.map(({ channel_url }) => channel_url !== undefined && urisToResolve.push(channel_url));
 
-  const displayedComments = fetchedReplies;
+    if (urisToResolve.length > 0) doResolveUris(urisToResolve);
+  }, [alreadyResolved, doResolveUris, fetchedReplies]);
 
-  return (
-    Boolean(numDirectReplies) && (
-      <div className="comment__replies-container">
-        {Boolean(numDirectReplies) && !isExpanded && (
-          <div className="comment__actions--nested">
-            <Button
-              className="comment__action"
-              label={__('Show Replies')}
-              onClick={() => setExpanded(!isExpanded)}
-              icon={isExpanded ? ICONS.UP : ICONS.DOWN}
-            />
-          </div>
-        )}
-        {isExpanded && (
-          <div>
-            <div className="comment__replies">
-              <Button className="comment__threadline" aria-label="Hide Replies" onClick={() => setExpanded(false)} />
+  // Wait to only display topLevelComments after resolved or else
+  // other components will try to resolve again, like channelThumbnail
+  React.useEffect(() => {
+    if (!isResolvingReplies) setCommentsToDisplay(fetchedReplies);
+  }, [isResolvingReplies, fetchedReplies]);
```

Comment:
## Note; not request for change (unless you feel otherwise)
```
  // Wait to only display topLevelComments after resolved or else
  // other components will try to resolve again, like channelThumbnail
```
The earlier suggestion of using the cache-aware `doResolveUris(..., true)` was for these components as well (any sub-components of `CommentList`).   I'll ask Jessop why we don't make the `true` as the default.

If we went that path, we:
- wouldn't need to filter a `resolvedComments[]`
- wouldn't need an extra `commentsToDisplay[]` state
- wouldn't need the extra effects and variables
All we need is just to `doResolveUris(topLevelComments.map(), true)` and other sub-components to use `true` as well.

Reply:
The problem was that using `topLevelComments` unresolved would make sub-components like `channelThumbnail` resolve each comment one by one first, and using `resolvedComments` would cause #61 

> Comments inadvertently masked